src/components/Link.tsx

```diff
@@ -44,121 +40,66 @@ export interface IProps extends IBaseProps, LinkProps {
+  let href = (to ?? hrefProp)!
 
-  const isDiscordInvite = url.isDiscordInvite(to)
-  if (isDiscordInvite) to = new URL(DISCORD_PATH, SITE_URL).href
-  const isExternal = url.isExternal(to)
-  const isHash = url.isHash(to)
-  const isStatic = url.isStatic(to)
-  const isPdf = url.isPdf(to)
-
-  const commonProps = {
-    dir,
-    ...restProps,
-  }
-
-  // Must use Chakra's native <Link> for anchor links
-  // Otherwise the Gatsby <Link> functionality will navigate to homepage
-  // See https://github.com/gatsbyjs/gatsby/issues/21909
-  // TODO: review the comments above for NextJS version
-  if (isHash) {
-    return (
-      <ChakraLink href={to} {...commonProps}>
-        {children}
-      </ChakraLink>
-    )
-  }
+  const isDiscordInvite = url.isDiscordInvite(href)
+  const isPdf = url.isPdf(href)
+  const isExternal = url.isExternal(href) || isPdf
 
   // Get proper download link for internally hosted PDF's & static files (ex: whitepaper)
   // Opens in separate window.
-  if ((isPdf && !isExternal) || isStatic) {
-    const relativePath = getRelativePath(router.asPath, to)
-
-    return (
-      <ChakraLink href={relativePath} isExternal {...commonProps}>
-        <>
-          {children}
-          <VisuallyHidden>(opens in a new tab)</VisuallyHidden>
-          {!hideArrow && (
-            <Icon
-              as={RxExternalLink}
-              boxSize="6"
-              p="1"
-              verticalAlign="middle"
-              me="-1"
-            />
-          )}
-        </>
-      </ChakraLink>
-    )
+  if (isPdf) {
+    href = getRelativePath(router.asPath, href)
   }
 
-  if (isExternal) {
-    return (
-      <ChakraLink href={to} isExternal {...commonProps}>
-        <>
-          {children}
-          <VisuallyHidden>(opens in a new tab)</VisuallyHidden>
-          {!hideArrow && (
-            <Icon
-              as={RxExternalLink}
-              boxSize="6"
-              p="1"
-              verticalAlign="middle"
-              me="-1"
-            />
-          )}
-        </>
-      </ChakraLink>
-    )
+  if (isDiscordInvite) {
+    href = new URL(DISCORD_PATH, SITE_URL).href
   }
 
   return (
-    <ChakraLink
-      to={to}
-      // as={IntlLink}
-      // language={language}
-      partiallyActive={isPartiallyActive}
-      activeStyle={activeStyle ? activeStyle : { color: theme.colors.primary }}
-      whiteSpace={"normal"}
-      {...commonProps}
-    >
+    // @ts-ignore: `isExternal` is missing from the NextLink types
+    <NextLink ref={ref} href={href} isExternal={isExternal} {...rest}>
       {children}
-    </ChakraLink>
+      {isExternal && <ExternalLinkContent hideArrow={hideArrow} />}
+    </NextLink>
   )
-}
-
-// export function navigate(to: string, language: Lang, options?: NavigateOptions<{}>) {
-//   if (typeof window === "undefined") {
-//     return
-//   }
+})
 
-//   const link = `/${language}${to}`
-//   gatsbyNavigate(link, options)
-// }
+const ExternalLinkContent = ({
```

Comment:
Right. 

I did this because I'm using the Chakra's Link integration component from `@chakra-ui/next-js` and it already takes care of external links. I don't see any error or warning when testing it.

However, I do see in the example page, that the external link has the `target="_blank"` but not the `rel="noopener"` attr anymore. That doesn't look great to me.

I'll change it back to use a plain Chakra Link 👍🏼